walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,72 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import com.google.sps.Event;
+import com.google.sps.TimeRange;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
+  private boolean eventParticipantInMeeting(Set<String> eventAttendees, 
+                                              Collection<String> meetingAttendees) {
+    for (String eventAttendee : eventAttendees) {
+      if (meetingAttendees.contains(eventAttendee)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> openMeetingTimes = new ArrayList<>();
+    List<Event> eventList = new ArrayList<>(events);
+
+    // Remove any events where there are no participants in event that are in meeting request.
+    for(Iterator<Event> it = eventList.iterator(); it.hasNext();) {
+      Event cur = it.next();
+      if (eventParticipantInMeeting(cur.getAttendees(), request.getAttendees()) == false) {
```

Comment:
consider to use filter() from Stream https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html

Reply:
Oh wow this is very nice, thanks for the tip! Done.